runhouse/resources/hardware/cluster.py

```diff
@@ -646,6 +645,32 @@ def keep_warm(self):
         )
         return self
 
+    def _is_cluster_shared(self) -> bool:
+        rns_address = self.rns_address
+        if rns_address is None:
+            return False
+
+        # If shared the creds used for launching will be attributed to another user
+        creds: str = (
+            self._resource_string_for_subconfig(self._creds, True)
+            if hasattr(self, "_creds") and self._creds
+            else None
+        )
+
+        if not creds or isinstance(creds, dict):
+            # specifying custom creds (for non on-demand clusters)
+            return False
+
+        cluster_folder = rns_client.base_folder(rns_address)
+        cluster_creds_folder = rns_client.base_folder(creds)
+
+        if cluster_folder != rns_client.username and (
+            creds and cluster_creds_folder != rns_client.username
```

Comment:
would `creds` ever equal `rns_client.username` or is just checking the folders sufficient? in `creds and cluster_creds_folder != rns_client.username`